tmt/steps/report/__init__.py

```diff
@@ -79,14 +83,18 @@ class ReportPlugin(tmt.steps.Plugin):
     how = 'display'
 
     # List of all supported methods aggregated from all plugins
-    _supported_methods = []
+    _supported_methods: List[tmt.steps.Method] = []
 
     @classmethod
-    def base_command(cls, method_class=None, usage=None):
+    def base_command(
+            cls,
+            method_class: Optional[Any] = None,
```

Comment:
What I'm sure about is that the use of `click.Option` in https://github.com/teemtee/tmt/pull/1327/ is likely wrong - that's what we realized last week or so on IRC, while looking into https://github.com/teemtee/tmt/pull/1304. What you're looking for there is `ClickOptionDecoratorType`, that's what all these `options()` methods return.

And since I now know which patch were we chatting about, I can point you to the right type, which should be `Type[click.Command]`, not `click.Command`. IIUIC, `method_class` is supposed to be exactly that, a class, not an instance, which means that you need one step back and "dereference" from "instance of click.Command class" to "type of instance of click.Command", which is done with `Type[...]` annotation.

Note that @thrix is adding this very annotation to `method_class` - mypy raised some complaints and this was needed to amend what `options.py` thins is right and what the rest of tmt code expected.

Reply:
> What I'm sure about is that the use of click.Option in https://github.com/teemtee/tmt/pull/1327 is likely wrong.

Ok, thanks for the clarification. That means that all the current `def options(...) -> List[click.Option]:` should be changed. I see it is annotated correctly in #1304 and the other occurrence I found is in [`provision/artemis.py`](https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/artemis.py#L309).

> I can point you to the right type, which should be Type[click.Command], not click.Command.

Yes, sorry, I wrote `click.Command` but I meant `Type[click.Command]` - I understand the difference.

Let's block this PR until #1304 is merged because this patch is depending on some of its changes.